samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -310,11 +310,17 @@ public void onBarrierStateChanged(final String version, ZkBarrierForVersionUpgra
         debounceTimer.scheduleAfterDebounceTime(barrierAction, 0, () -> onNewJobModelConfirmed(version));
       } else {
         if (ZkBarrierForVersionUpgrade.State.TIMED_OUT.equals(state)) {
-          // no-op
-          // In our consensus model, if the Barrier is timed-out, then it means that one or more initial
-          // participants failed to join. That means, they should have de-registered from "processors" list
-          // and that would have triggered onProcessorChange action -> a new round of consensus.
-          LOG.info("Barrier for version " + version + " timed out.");
+          // no-op for non-leaders
+          // for leader: make sure we do not stop - so generate a new job model
+          LOG.warn("Leader: barrier for version " + version + " timed out.");
```

Comment:
Minor suggestion(log message format): 

s/Leader: barrier for version /Barrier for version. 

At this point, we're not sure if it's leader or not.